spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java

```diff
@@ -1186,8 +1188,7 @@ public void setObservationEnabled(boolean observationEnabled) {
 	 * @param observationConvention the convention.
 	 * @since 3.0
 	 */
-	public void setObservationConvention(RabbitListenerObservationConvention observationConvention) {
-		Assert.notNull(observationConvention, "'observationConvention' cannot be null");
+	public void setObservationConvention(ObservationConvention<RabbitMessageReceiverContext> observationConvention) {
```

Comment:
I'm not sure how is that harmful to expect a generic convention, but in Spring Integration I extract an interface like this: https://github.com/spring-projects/spring-integration/pull/3879/files#diff-ae850c6c6d3c04d681a2509ef10dae6386011a072e47e761be4e320b3245539aR32.
Pay attention to that `supportsContext()` method impl.

@marcingrzejszczak , what is the proper path to go?

Reply:
That makes sense.